pyteal/ast/if_.py

```diff
@@ -25,17 +26,26 @@ def __init__(self, cond: Expr, thenBranch: Expr, elseBranch: Expr = None) -> Non
         """
         super().__init__()
         require_type(cond.type_of(), TealType.uint64)
+        # Flag to denote and check whether the new If().Then() syntax is being used
+        self.alternateSyntaxFlag = False
 
-        if elseBranch is None:
-            require_type(thenBranch.type_of(), TealType.none)
+        if thenBranch:
+            if elseBranch:
+                require_type(thenBranch.type_of(), elseBranch.type_of())
+            else:
+                # If there is only a thenBranch, then it should evaluate to none type
+                require_type(thenBranch.type_of(), TealType.none)
         else:
-            require_type(thenBranch.type_of(), elseBranch.type_of())
+            self.alternateSyntaxFlag = True
         
         self.cond = cond
         self.thenBranch = thenBranch
         self.elseBranch = elseBranch
 
     def __teal__(self, options: 'CompileOptions'):
+        if self.thenBranch is None:
+            raise TealCompileError("If expression must have a thenBranch", self.thenBranch)
```

Comment:
The second arg to `TealCompileError` should just be `self`, since `self.thenBranch` is None at this point.

Same thing for lines 70 and 77.

Reply:
Will fix, then push!